adapter/ipListChecker/adapter.go

```diff
@@ -15,74 +15,93 @@
 package ipListChecker
```

Comment:
Giving this another look over, I now realize that the new design completely foregoes adapter-level config, all we have is an aspect level configuration.

I really believe there is a need for global adapter-level config, distinct from aspect-level config. As I mentioned previously, we use this kind of stuff all the time in Google. You can supply command-line options to a server when it starts up which is global to the server, independent of any configuration you may give to the server for individual tenants.

Since adapters don't have command-line options, adapter-level config would be the logical equivalent.